imageregistry/v1/types.go

```diff
@@ -303,8 +302,60 @@ type ImageRegistryConfigStorageAzure struct {
 	// object.
 	// +optional
 	CloudName string `json:"cloudName,omitempty"`
+	// networkAccess defines the network access properties for the storage account.
+	// type: Internal means the storage account will be private, type: External
+	// means the storage account will be publicly accessible.
+	// When type: Internal is used, a vnetName, subNetName and privateEndpointName
+	// may optionally be specified. If unspecificed, the image registry operator
+	// will discover vnet and subnet names, and generate a privateEndpointName.
+	// Defaults to type: External.
+	// +kubebuilder:default={"type": "External"}
+	// +optional
+	NetworkAccess *AzureNetworkAccess `json:"networkAccess,omitempty"`
+}
+
+type AzureNetworkAccess struct {
+	// type is the network access level to be used for the storage account.
+	// type: Internal means the storage account will be private, type: External
+	// means the storage account will be publicly accessible.
+	// Defaults to "External".
+	Type AzureNetworkAccessType `json:"type,omitempty"`
+	// internal defines the vnet and subnet names to configure a private
+	// endpoint and connect it to the storage account in order to make it
+	// private.
+	// when type: Internal and internal is unset, the image registry operator
+	// will discover vnet and subnet names, and generate a private endpoint
+	// name.
+	// +optional
+	Internal *AzureNetworkAccessInternal `json:"internal,omitempty"`
 }
 
+type AzureNetworkAccessInternal struct {
+	// vnetName is the name of the vnet the registry operates in.
+	// +kubebuilder:validation:MaxLength=64
+	// +kubebuilder:validation:MinLength=2
+	// +optional
+	VNetName string `json:"vnetName,omitempty"`
+	// subnetName is the name of the subnet the registry operates in.
+	// +kubebuilder:validation:MaxLength=80
+	// +kubebuilder:validation:MinLength=1
+	// +optional
+	SubnetName string `json:"subnetName,omitempty"`
+	// privateEndpointName is the name of the private endpoint for the registry.
+	// +kubebuilder:validation:MaxLength=64
+	// +kubebuilder:validation:MinLength=2
+	// +kubebuilder:validation:Pattern=`^[0-9A-Za-z][0-9A-Za-z_.-]*[0-9A-Za-z_]$`
+	// +optional
+	PrivateEndpointName string `json:"privateEndpointName,omitempty"`
+}
+
+type AzureNetworkAccessType string
+
+const (
+	AzureNetworkAccessTypeInternal AzureNetworkAccessType = "Internal"
+	AzureNetworkAccessTypeExternal AzureNetworkAccessType = "External"
```

Comment:
// AzureNetworkAccessTypeExternal means the storage account will be publicly accessible

Reply:
Added